transports/httpclient/src/main/java/org/commonjava/maven/galley/transport/htcli/internal/HttpDownload.java

```diff
@@ -147,8 +148,23 @@ private void writeTarget()
             }
             catch ( final IOException e )
             {
-                throw new TransferException( "Failed to write to local proxy store: {}\nOriginal URL: {}. Reason: {}",
-                                             e, target, url, e.getMessage() );
+                ConcreteResource resource = null;
+                if ( target.exists() )
+                {
+                    try
+                    {
+                        resource = target.getResource();
+                        target.delete();
+                    }
+                    catch ( IOException e1 )
+                    {
+                        throw new TransferException(
```

Comment:
Log this to the error output too, and make sure you log the original IOException to error too so we don't lose it. I agree this will be the higher priority error and needs to be thrown instead of TransferContentException, but we need to ensure the original error information is in the log. In fact, I'd also include the original error.getMessage() in the TransferException too.

Reply:
ok